infra/templates/az-isolated-service-single-region/docs/design.md

```diff
@@ -0,0 +1,169 @@
+
+This document outlines how Cobalt can be extended to meet the use cases of these customers. The intended audience of this document is the development and product teams working on Cobalt and related projects. 
+
+## In Scope 
+
+- Identify deployment topology needed by the customer 
+- Identify key Terraform templates needed for deployment
+- Identify key Terraform modules needed for deployment
+- Identify gaps in Terraform provider templates
+
+## Out of scope 
+
+- External customer sign-off
+- Template (Terraform, ARM) implementation
+
+ 
+
+## Key Terms 
+- **RG**: Abbreviation for “Resource Group” 
+- **Sub**: Abbreviation for “Subscription” 
+- **Persona**: An archetype of a Cobalt customer 
+- **Stage**: An application deployment stage (dev, qa, pre-prod, prod, etc...) 
+- **Region**: A location in which an application is deployed 
+
+ 
+
+## Customers 
+- **Admin**: This persona represents an administrator of Azure. This persona does not implement the line of business applications but will help other teams deliver them. 
+- **App Developer Team**: This persona is responsible for creating and maintaining the line of business applications 
+
+## Open Questions 
+
+These open questions may impact the implementation plan and overall scope of the deployment described below 
+
+- What is the clients use case for Cosmos? Who manages these instances, and should we be automating the creation of them? Do we create the RG hosting Cosmos? If not, who owns this?
+
+## Deployment Topology 
+
+This graphic shows the targeted deployment topology needed by our enterprise customers. The deployment is split across 3 subscriptions, each subscription belonging to a single tenant. The resources are partitioned to align with the different personas within the customer – subscription 1 & 3 are where admin resources exist, whereas subscription 2 hosts the application developer team resources. 
+
+![Deployment Topology](./.design_images/deployment_topology.png "Deployment Topology")
+
+## Template Topology 
+
+The graphic below outlines the topology of the terraform templates that will deploy the topology called out above. By leveraging a multi-subscription deployment via [provider aliasing](https://www.terraform.io/docs/configuration/providers.html#alias-multiple-provider-instances) we can use a single template to target multiple Azure subscriptions.
+
+![Template Topology](./.design_images/template_topology.png "Template Topology")
+
+
+## Question & Answer 
+
+> Why not split the deployment along the subscription boundary? Doing so will maintain a 1-1 mapping between template and Azure subscription and it naturally splits along the same structure as our customer’s organization? 
+
+Having a single template reduces the coordination required to manage deployment orchestration and secrets management. While this may be useful for more complex cases, it will need a design of its own as it requires some hard thinking about deployment orchestration and composition. 
```

Comment:
You may also want to add a question on whether you can have two TF templates run within the same workspace? one for the network admins and another for the app dev team to provision service plan scaling rules or monitoring settings.